glean-core/src/metrics/labeled.rs

```diff
@@ -338,8 +278,8 @@ pub fn validate_dynamic_label(
         );
         record_error(glean, meta, ErrorType::InvalidLabel, msg, None);
         true
-    } else if !matches_label_regex(label) {
-        let msg = format!("label must be snake_case, got '{}'", label);
+    } else if !label.is_ascii() {
```

Comment:
I was about to suggest something like `.is_printable` but apparently that doesn't exists.
There's only [`char::is_ascii_control`](https://doc.rust-lang.org/std/primitive.char.html#method.is_ascii_control),
so this would need to be:

```suggestion
    } else if label.chars().any(|c| c.is_ascii_control()) {
```

Note-worth: this allows spaces, but I guess that's fine.